build-tools-internal/src/integTest/groovy/org/elasticsearch/gradle/internal/snyk/SnykDependencyMonitoringGradlePluginFuncTest.groovy

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal.snyk
+
+import org.elasticsearch.gradle.fixtures.AbstractGradleInternalPluginFuncTest
+import org.elasticsearch.gradle.fixtures.http.HttpServerRule
+import org.elasticsearch.gradle.internal.conventions.precommit.LicenseHeadersPrecommitPlugin
+import org.gradle.api.Plugin
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.Rule
+import org.skyscreamer.jsonassert.JSONAssert
+
+class SnykDependencyMonitoringGradlePluginFuncTest extends AbstractGradleInternalPluginFuncTest {
+
+    @Rule
+    public HttpServerRule httpServer = new HttpServerRule()
+
+    Class<? extends Plugin> pluginClassUnderTest = SnykDependencyMonitoringGradlePlugin.class
+
+    def setup() {
+        configurationCacheCompatible = false // configuration is not cc compliant
```

Comment:
For creating the dependency graph we need to work on the `Configuration` object that is not compliant with the gradle configuration-cache at the moment. This should not affect our developers as this task should never be triggered in day to day life from dev environments